tests/test_set_config.py

```diff
@@ -217,13 +217,12 @@ def test_incompatible_file_type(): #pylint: disable=unused-variable
     input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/model_configure.sample"))
     args = ['-i', input_file]
 
-    outstring = io.StringIO()
-    with redirect_stdout(outstring):
+    with pytest.raises(ValueError) as error:
         set_config.create_config_obj(args)
-    result = outstring.getvalue()
-    outcome = "Set config failure: bad file type\n"
 
-    assert result == outcome
+        expected = "Set config failure: bad file type\n"
+        actual = str(error.value)
+        assert actual == expected
```

Comment:
Here, I think just calling set_config.create_config_obj() with bad args would be the only thing needed for the context manager to catch the expected Exception.